UI/ChatGPT/ChatGPT/ChatGPT/Presentation/MainModel.cs

```diff
@@ -0,0 +1,82 @@
+using ChatGPT.Business;
+using ChatGPT.Extensions;
+using ChatGPT.Services;
+
+namespace ChatGPT.Presentation;
+
+public partial record MainModel
+{
+	private readonly IChatService _chatService;
+	public MainModel(IChatService chatService)
+	{
+		_chatService = chatService;
+	}
+
+	public IState<string> Prompt => State.Value(this, () => string.Empty);
+
+	public IListState<Message> Messages => ListState<Message>.Empty(this);
+
+	public IState<bool> IsMessageStream => State.Value(this, () => !OperatingSystem.IsBrowser());
+
+	public IState<bool> IsStreamEnabled => State.Value(this, () => !OperatingSystem.IsBrowser());
+
+	public async ValueTask AskMessage(string prompt, CancellationToken ct)
+	{
+		if (await IsMessageStream)
+		{
+			await AskAsStream(prompt, ct);
+		}
+		else
+		{
+			await Ask(prompt, ct);
+		}
+	}
+
+	public async ValueTask Ask(string prompt, CancellationToken ct)
```

Comment:
We can make it private

```suggestion
	private async ValueTask Ask(string prompt, CancellationToken ct)
```

Reply:
Done in next commit.